core/geometry/src/polyface/Polyface.ts

```diff
@@ -265,20 +262,20 @@ export class IndexedPolyface extends Polyface {
     }
 
     // Add color and color index data
-    if (this.data.color && source.data.color && source.data.colorIndex) {
-      const startOfNewColors = this.data.color.length;
-      for (const sourceColor of source.data.color) {
+    if (copyColors) {
```

Comment:
Can you replace ! with assert here?

Reply:
Same for copyColors (see previous comment).